### PR TITLE
client: expose network namespace CNI config as task env vars.

### PR DESCRIPTION
**Pull request by jrasell**
I forgot I had this stashed locally. Raising it initially as a draft PR
to get early feedback and will modify and add docs etc at a later
date.

This change exposes CNI configuration details of a network
namespace as environment variables. This allows a task to use
these value to configure itself; a potential use case is to run
a Raft application binding to IP and Port details configured using
the bridge network mode.

### CODE REVIEW

File: client/taskenv/env.go [hunk 1]
```diff
@@ -939,6 +961,18 @@ func buildUpstreamsEnv(envMap map[string]string, upstreams []structs.ConsulUpstr
 	}
 }
 
+// addNetNamespacePort builds NOMAD_NET_NAMESPACE_{IP,PORT,ADDR}_{port_label}
```
Comment (tgross) [inline client/taskenv/env.go:964]: As written, I think it's `NOMAD_ALLOC_{INTERFACE,IP,PORT,ADDR}_{port_label}`, but maybe `NOMAD_NET_NAMESPACE_*` would be better?

**Review by tgross — COMMENTED** [review]
The approach here seems technically sounds. We have an existing set of `NOMAD_IP`, etc. variables described in the [`network`](https://www.nomadproject.io/docs/job-specification/network) docs... how do we envision these interacting?

**Comment by jrasell** [comment]
> The approach here seems technically sound. We have an existing set of `NOMAD_IP`, etc. variables described in the [`network`](https://www.nomadproject.io/docs/job-specification/network) docs... how do we envision these interacting?

iirc these represent the host assignments when running in `network.mode = "host"`. When I have a moment, I'll double check this.

**Comment by sundbry** [comment]
I've just tested this in a CNI network using calico. It works great and saved a bunch of hassle modding container images to autodetect their IP. Thanks @jrasell.

**Review by tgross — APPROVED** [review]
LGTM!

**Pull request merged**